src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1396,6 +1402,29 @@ internal static bool OnServerCertificateCustomValidationCallback(object sender,
             return false;
         }
 
+        private static void ThrowNameMismatchException(object sender, X509Certificate cert)
+        {
+            var hostname = ((HttpRequestMessage)sender).RequestUri.DnsSafeHost;
+#if NETSTANDARD1_3
+            var cert2 = new X509Certificate2(cert.Handle);
```

Comment:
var cert2 = cert as X509Certificate2 ?? new X509Certificate2(cert.Handle);

It looks like not much is happening in ctor, but let's do this.